minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java

```diff
@@ -257,8 +257,10 @@ private void startMaster(TachyonConf testConf) throws IOException {
     // TODO(andrew): Move logic to the integration-tests project so that we can use instanceof here
     // instead of comparing classnames.
     if (ufs.getClass().getSimpleName().equals("LocalMiniDFSCluster")) {
-      testConf.set(Constants.UNDERFS_ADDRESS, ufs.getUnderFilesystemAddress() + mTachyonHome);
-      MasterContext.reset(testConf);
+      String ufsAddress = ufs.getUnderFilesystemAddress() + mTachyonHome;
+      testConf.set(Constants.UNDERFS_ADDRESS, ufsAddress);
+      MasterContext.getConf().set(Constants.UNDERFS_ADDRESS, ufsAddress);
+      mMasterConf = MasterContext.getConf();
```

Comment:
I didn't want to rely on the implementation of `MasterContext.reset` resetting by reference instead of copying `mMasterConf`. Also, this is needed if `MasterConf` gets reset anywhere in `mMaster.create` or `mMaster.start`. None of this is the case now, but it could change in the future so I think it's better to be explicit.